README/RCodingStyle.hxx

```diff
@@ -0,0 +1,251 @@
+   // This is not enforced by C++ but required by this coding style.
+   using Nesteds_t = std::vector<Nested>;
+
+   // THIRD section: data members.
+
+   // Data member names start with `f`.
+   // Documentation comments can be in front of data members or right behind them, see below.
+   // For the latter case, comments must be column-aligned:
+
+   int fIntMember = -2;                  ///< Use inline initialization at least for fundamental types.
+   std::vector<Nested> fManyStrings;     ///<! This marks a member that does not get serialized (I/O comment).
+   static constexpr const int kSeq = -7; ///< A static constexpr (const) variable starts with `k`.
+
+   /// Don't explicitly default initialize (`{}`)
+   std::vector<ROOT::Experimental::RAxisConfig> fAxes; //! For pre-comment cases, ROOT I/O directives go here.
+
+   /// A static variable starts with `fg`. Avoid all non-constexpr static variables! Long data
+   /// member documentation is just fine, but then put it in front of the data member.
+   static std::string fgNameBadPleaseUseStaticFunctionStaticVariable;
+
+protected:
+   /// FOURTH section: non-public functions.
+
+   /// Instead of static data members (whether private or public), use outlined static
+   /// functions. See the implementation in the source file.
+   static const std::string &AccessStaticVar();
+
+public:
+   // FIFTH section: methods, starting with constructors, then alphabetical order, possibly
+   // grouped by functionality (\ingroup or \{ \}).
+
+   /// Use `= default` inside the class whenever possible
+   RExampleClass() = default;
+
+   /// Classes with virtual functions must have a virtual destructor. If all other functions are
+   /// pure virtual or inlined, the destructor should be outlined to pin the class's vtable to an
+   /// object file. I.e. say `= default` in the source, not here.
+   virtual ~RExampleClass();
+
+   /// For trivial functions, use `const` and possibly `constexpr`, but not `noexcept`.
+   int GetAsInt() const { return fIntMember; }
+
+   /// Static functions line up alphabetically with the others.
+   static int GetSeq() { return kSeq; }
+
+   /// It is fine to have one-line function declarations.
+   void SetInt(int i) { fIntMember = i; }
+   
+   /// \brief Replace the many strings
+   /// \param[in] input A collection of strings to use.
+   /// \return Always return true but if it could fail would return false in that case.
+   ///
+   /// All functions must have at least one line of comments, parseable/useable by doxygen.
+   bool SetManyStrings(const std::vector<Nested>& input) { fManyStrings = input; return true; }
+
+   /// Virtual functions are still fine (though we try to avoid them as they come with a call
+   /// and optimization penalty). See `RCodingStyle` for overriding virtual functions.
+   virtual void VirtualFunction() = 0;
+};
+
+///\class RCodingStyle
+/// This is exhaustive documentation of the class template, explaining the constraints
```

Comment:
**The first thing after the \class macro** should be the brief documentation. Doxygen takes the first sentence of the docs if no `\brief` is passed, so try to just write one sentence in one line that ends with a `.`.